tests/integration/Core/Content/Category/SalesChannel/NavigationRouteTest.php

```diff
@@ -239,6 +246,192 @@ public function testInclude(): void
+            [
+                'id' => $landingPageId,
+                'name' => 'Test Landing Page',
+                'url' => 'test-landing-page',
+                'active' => true,
+                'salesChannels' => [
+                    ['id' => $this->ids->get('sales-channel')],
+                ],
+            ],
+        ], Context::createDefaultContext());
+
+        $this->getContainer()->get('category.repository')->update([
+            [
+                'id' => $this->ids->get('category3'),
+                'type' => CategoryDefinition::TYPE_LINK,
+                'linkType' => CategoryDefinition::LINK_TYPE_LANDING_PAGE,
+                'internalLink' => $landingPageId,
+            ],
+        ], Context::createDefaultContext());
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category3') && $category['linkType'] === CategoryDefinition::LINK_TYPE_LANDING_PAGE) {
+                static::assertStringContainsString('test-landing-page', $category['seoLink']);
+            }
+        }
+
+        $this->createSeoUrl(
+            'frontend.landing.page',
+            '/landingPage/' . $landingPageId,
+            'custom-landing-page-url',
+            $landingPageId
+        );
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category3') && $category['linkType'] === CategoryDefinition::LINK_TYPE_LANDING_PAGE) {
+                static::assertStringContainsString('/custom-landing-page-url', $category['seoLink']);
+            }
+        }
+    }
+
+    public function testProductInternalLinkHasSeoUrl(): void
+    {
+        $productId = Uuid::randomHex();
+        $productBuilder = new ProductBuilder($this->ids, 'TEST-1234');
+        $productBuilder->id = $productId;
+        $productBuilder
+            ->name('Test Product')
+            ->price(15, 10)
+            ->visibility($this->ids->get('sales-channel'), ProductVisibilityDefinition::VISIBILITY_ALL)
+            ->active(true);
+
+        $productBuilder->write($this->getContainer());
+
+        $this->getContainer()->get('category.repository')->update([
+            [
+                'id' => $this->ids->get('category3'),
+                'type' => CategoryDefinition::TYPE_LINK,
+                'linkType' => CategoryDefinition::LINK_TYPE_PRODUCT,
+                'internalLink' => $productId,
+            ],
+        ], Context::createDefaultContext());
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        foreach ($response as $category) {
+            if ($category['id'] === $this->ids->get('category3') && $category['linkType'] === CategoryDefinition::LINK_TYPE_PRODUCT) {
+                static::assertStringContainsString('/Test-Product/', $category['seoLink']);
```

Comment:
You can break out of the loop if the right category was found to reduce the number of iterations